From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: celenity <celenity@celenity.dev>
Date: Sun, 05 Jul 2026 20:19:20 +0000
Subject: [PATCH] Gecko - Disable Nimbus

Disables and prevents A/B testing/experimentation and remote configuration of settings.

Signed-off-by: celenity <celenity@celenity.dev>
---
diff --git a/toolkit/components/nimbus/ExperimentAPI.sys.mjs b/toolkit/components/nimbus/ExperimentAPI.sys.mjs
index b7a013c9b57e..660a8a705313 100644
--- a/toolkit/components/nimbus/ExperimentAPI.sys.mjs
+++ b/toolkit/components/nimbus/ExperimentAPI.sys.mjs
@@ -36,7 +36,7 @@ const CRASHREPORTER_ENABLED =
   AppConstants.MOZ_CRASHREPORTER && AppConstants.MOZ_APP_NAME !== "thunderbird";
 
 const IS_MAIN_PROCESS =
-  Services.appinfo.processType === Services.appinfo.PROCESS_TYPE_DEFAULT;
+  false;
 
 const Prefs = Object.freeze({
   AI_FEATURES_ENABLED: "browser.ai.control.default",
@@ -536,6 +536,9 @@ export const ExperimentAPI = new (class {
    *          store
    */
   async ready() {
+    if (!this.#initializedPromise) {
+      return Promise.resolve();
+    }
     return this.manager.store.ready();
   }
 
diff --git a/toolkit/components/nimbus/lib/ExperimentManager.sys.mjs b/toolkit/components/nimbus/lib/ExperimentManager.sys.mjs
index f7c6421ebcd7..949543a1f979 100644
--- a/toolkit/components/nimbus/lib/ExperimentManager.sys.mjs
+++ b/toolkit/components/nimbus/lib/ExperimentManager.sys.mjs
@@ -78,7 +78,7 @@ const CannotEnrollFeatureReason = Object.freeze({
  */
 
 const IS_MAIN_PROCESS =
-  Services.appinfo.processType === Services.appinfo.PROCESS_TYPE_DEFAULT;
+  false;
 
 export const UnenrollmentCause = {
   fromCheckRecipeResult(result) {
@@ -236,7 +236,7 @@ export class ExperimentManager {
         return new Date();
       },
     };
-    Object.defineProperty(context, "activeExperiments", {
+/*    Object.defineProperty(context, "activeExperiments", {
       enumerable: true,
       get: async () => {
         await this.store.ready();
@@ -286,7 +286,7 @@ export class ExperimentManager {
           return acc;
         }, {});
       },
-    });
+    }); */
     return context;
   }
 
-- 
